src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -34,6 +34,26 @@ public class PooledQueueCache<TQueueMessage, TCachedMessage>
         private readonly ICacheDataAdapter<TQueueMessage, TCachedMessage> cacheDataAdapter;
         private readonly ICacheDataComparer<TCachedMessage> comparer;
 
+        public TCachedMessage? Newest
+        {
+            get
+            {
+                if (IsEmpty)
+                    return null;
+                return messageBlocks.First.Value.NewestMessage;
```

Comment:
this is probably not multi-threaded, but just in case (and also reads easier) now that we use c# 6 you can write:
`return messageBlocks.First?.Value.NewestMessage;`
without the need to check for empty or null